src/openboardview/FileFormats/BRD2File.cpp

```diff
@@ -218,21 +218,7 @@ BRD2File::BRD2File(std::vector<char> &buf) {
 		parts.push_back(part);
 	}
 
-	for (auto &nail : nails) {
-		BRDPin pin;
-		pin.pos = nail.pos;
-		if (nail.side == 1) {
-			pin.part = parts.size();
-			pin.side = BRDPinSide::Top;
-		} else {
-			pin.pos.y = max.y - pin.pos.y;
-			pin.part  = parts.size() - 1;
-			pin.side = BRDPinSide::Bottom;
-		}
-		pin.probe = nail.probe;
-		pin.net   = nail.net;
-		pins.push_back(pin);
-	}
+	AddNailsAsPins(max);
```

Comment:
I don't even remember why this piece of code was there. Is it actually useful to be extracted outside of this format?

Reply:
Yes, it is used for vias in gencad format - see https://github.com/galkinvv/OpenBoardView/blob/474fc19a84fdef92aee3dea7c8c873c2c3b63288/src/openboardview/FileFormats/GenCADFile.cpp#L95

So extracted to base calss to avoid duplication.